File: app/src/main/java/dolphin/android/util/DateUtils.java
package dolphin.android.util;

import java.util.Calendar;

/**
 * Created by dolphin on 2017/05/20.
 * <p>
 * Quick access to some comparisons.
 */

public class DateUtils {
    public static boolean sameDay(Calendar cal1, Calendar cal2) {
        return cal1.get(Calendar.YEAR) == cal2.get(Calendar.YEAR) &&
                cal1.get(Calendar.DAY_OF_YEAR) == cal2.get(Calendar.DAY_OF_YEAR);
    }

    public static boolean isToday(Calendar calendar) {
        return sameDay(Calendar.getInstance(), calendar);
    }
}

File: app/src/main/java/dolphin/android/util/DateUtils.kt
package dolphin.android.util

import java.util.*

/**
 * Quick access to some comparisons.
 */

class DateUtils {
    companion object {
        @JvmStatic
        fun sameDay(cal1: Calendar, cal2: Calendar): Boolean {
            return cal1.get(Calendar.YEAR) == cal2.get(Calendar.YEAR) &&
                    cal1.get(Calendar.DAY_OF_YEAR) == cal2.get(Calendar.DAY_OF_YEAR)
        }

        @JvmStatic
        fun isToday(calendar: Calendar): Boolean {
            return sameDay(Calendar.getInstance(), calendar)
        }
    }
}
